fhttp/http.go

```diff
@@ -750,9 +752,81 @@ var (
 	EchoRequests int64
 )
 
+// Format: status=503 for 100% 503s
+// status=503:20,404:10 for 20% 503s, 10% 404s, 70% 200s
+func parseStatus(status string) int {
+	lst := strings.Split(status, ",")
+	log.Debugf("Parsing status %s -> %v", status, lst)
+	// Simple non probabilistic status case:
+	if len(lst) == 1 && !strings.ContainsRune(status, ':') {
+		s, err := strconv.Atoi(status)
+		if err != nil {
+			log.Warnf("Bad input status %v, not a number nor coma and colon separated %% list", status)
+			return http.StatusBadRequest
+		}
+		log.Debugf("Parsed status %s -> %d", status, s)
+		return s
+	}
+	probTable := make([]int, 100)
+	for i := len(probTable) - 1; i >= 0; i-- {
+		probTable[i] = http.StatusOK
+	}
+	lastPercent := 0
+	for _, entry := range lst {
+		l2 := strings.Split(entry, ":")
+		if len(l2) != 2 {
+			log.Warnf("Should have exactly 1 : in status list %s -> %v", status, entry)
+			return http.StatusBadRequest
+		}
+		s, err := strconv.Atoi(l2[0])
+		if err != nil {
+			log.Warnf("Bad input status %v -> %v, not a number before colon", status, l2[0])
+			return http.StatusBadRequest
+		}
+		percStr := l2[1]
+		p, err := strconv.Atoi(percStr)
+		if err != nil || p < 0 || p > 100 {
+			log.Warnf("Percentage is not a [0 - 100] number in %v -> %v : %v %d", status, percStr, err, p)
+			return http.StatusBadRequest
+		}
+		if lastPercent+p > 100 {
+			log.Warnf("Sum of percentage is greater than 100 in %v %d %d", status, lastPercent, p)
+			return http.StatusBadRequest
+		}
+		for i := 0; i < p; i++ {
+			probTable[lastPercent+i] = s
+		}
+		lastPercent += p
+	}
+	res := rand.Intn(100)
+	log.Debugf("[0-100[ for %s roll got %d -> %d", status, res, probTable[res])
```

Comment:
do we need a full 100 entry array for this?

it seems like we could track this with a map of `len(lst)` with additive indices. we expect the number of status entries to be small and so running through the list each time is probably efficient enough.  this would also allow arbitrary precision on the percentages.

something like:
```
// for ?status=503:20,404:10

table := map[int]int{ 20:503, 30:404 }
res := rand.Intn(100)
for percent, code := range table {
  if res < percent {
     return code
  }
}
return http.StatusOK
```

not sure this matters all that much, but thought I'd ask.

Reply:
or, maybe we can do this as we parse status codes and avoid any secondary storage altogether?  generate the number first, walk through as we go... (haven't thought through all the error conditions in this approach, however).